autopilot/contractorspending.go

```diff
@@ -13,23 +14,43 @@ func (c *contractor) currentPeriod() uint64 {
 	return c.currPeriod
 }
 
-func (c *contractor) updateCurrentPeriod() {
-	c.mu.Lock()
-	defer func(prevPeriod uint64) {
-		if c.currPeriod != prevPeriod {
-			c.logger.Debugf("updated current period, %d->%d", prevPeriod, c.currPeriod)
-		}
-		c.mu.Unlock()
-	}(c.currPeriod)
+func (c *contractor) updateCurrentPeriod(ctx context.Context) error {
+	// check if we are synced
+	state := c.ap.state
+	if !state.cs.Synced {
+		return errors.New("can't update current period if we are not synced")
+	}
 
-	cfg := c.ap.state.cfg
-	cs := c.ap.state.cs
+	// fetch autopilot
+	autopilot, err := c.ap.bus.Autopilot(ctx, c.ap.id)
```

Comment:
Hm I'm not sure I like the naming here. We aren't really fetching an autopilot but rather the persistence or saved state or whatever we want to call it.